File: javascript - Copy/arraymenudriven.js
const readlineSync = require('readline-sync');

let n = parseInt(readlineSync.question("Enter the size of the array: "));
let arr = [];

console.log("Enter the elements of the array: ");
for (let i = 0; i < n; i++) {
    arr.push(parseInt(readlineSync.question("Element " + (i + 1) + ": ")));
}

while (n != 0) {
    console.log("\n1. Traversal/Linear Search\n2. String Reversal\n3. Updating the Element\n4. Inserting an Element at Specified Position\n5. Deleting an Element from Specified Position\n6. Sorting the Array (Bubble Sort)\n0. to exit\n");
    let choice = parseInt(readlineSync.question("Enter your choice: "));

    switch (choice) {
        case 0:
            console.log("Exiting program.");
            return;

        case 1:
            let searchElement = parseInt(readlineSync.question("Enter element to search: "));
            let found = false;
            for (let i = 0; i < n; i++) {
                if (arr[i] === searchElement) {
                    console.log(`The element ${searchElement} is present at index ${i + 1}`);
                    found = true;
                    break;
                }
            }
            if (!found) {
                console.log("Element not found.");
            }
            break;

        case 2:
            console.log("Reversed array: " + arr.reverse().join(' '));
            break;

        case 3:
            arr = arr.map((element, index) => index % 2 === 0 ? element * 2 : element + 5);
            console.log("Updated array: " + arr.join(' '));
            break;

        case 4:
            let insertPosition = parseInt(readlineSync.question("Enter position to insert: "));
            if (insertPosition < 1 || insertPosition > n + 1) {
                console.log("Invalid position.");
            } else {
                let insertElement = parseInt(readlineSync.question("Enter element to insert: "));
                arr.splice(insertPosition - 1, 0, insertElement);
                n++;
                console.log("Element inserted successfully.");
            }
            break;

        case 5:
            let deletePosition = parseInt(readlineSync.question("Enter position to delete: "));
            if (deletePosition < 1 || deletePosition > n) {
                console.log("Invalid position.");
            } else {
                arr.splice(deletePosition - 1, 1);
                n--;
                console.log("Element deleted successfully.");
            }
            break;

        case 6:
            for (let i = 0; i < n - 1; i++) {
                for (let j = 0; j < n - i - 1; j++) {
                    if (arr[j] > arr[j + 1]) {
                        let temp = arr[j];
                        arr[j] = arr[j + 1];
                        arr[j + 1] = temp;
                    }
                }
            }
            console.log("Sorted array: " + arr.join(' '));
            break;

        default:
            console.log("Invalid choice.");
    }
}

File: c++ - Copy/arraymenudriven.cpp
#include <iostream>
using namespace std;

int main() {
    int n, choice, i, s, pos;
    cout << "Enter the size of the array: ";
    cin >> n;
    int arr[n];
    cout << "Enter the elements of the array: ";
    for (i = 0; i < n; i++) {
        cin >> arr[i];
    }

    while (n != 0) {
        cout << "\n1. Traversal/Linear Search\n2. String reversal\n3. Updating the elements\n4. Inserting an element at specified position\n5. Deleting an element from specified position\n6. Sorting the array (Bubble Sort)\n0. To exit\n";
        cout << "Enter your choice: ";
        cin >> choice;

        switch (choice) {
            case 0:
                cout << "Exiting program.\n";
                return 0;

            case 1:
                cout << "Enter element to search ";
                cin >> s;
                for (i = 0; i < n; i++) {
                    if (arr[i] == s) {
                        cout << "The element " << s << " is present at index " << i + 1 << endl;
                    }
                }
                break;

            case 2:
                cout << "Reversed array: ";
                for (i = n - 1; i >= 0; i--) {
                    cout << arr[i] << " ";
                }
                cout << endl;
                break;

            case 3:
                for (i = 0; i < n; i++) {
                    if (i % 2 == 0) {
                        arr[i] *= 2;
                    } else {
                        arr[i] += 5;
                    }
                }
                cout << "Updated array: ";
                for (i = 0; i < n; i++) {
                    cout << arr[i] << " ";
                }
                cout << endl;
                break;

            case 4:
                cout << "Enter position to insert: ";
                cin >> pos;
                if (pos < 1 || pos > n + 1) {
                    cout << "Invalid position.\n";
                } else {
                    cout << "Enter element to insert: ";
                    cin >> s;
                    for (i = n - 1; i >= pos - 1; i--) {
                        arr[i + 1] = arr[i];
                    }
                    arr[pos - 1] = s;
                    n++;
                    cout << "Element inserted successfully.\n";
                }
                break;

            case 5:
                cout << "Enter position to delete: ";
                cin >> pos;
                if (pos < 1 || pos > n) {
                    cout << "Invalid position.\n";
                } else {
                    for (i = pos - 1; i < n - 1; i++) {
                        arr[i] = arr[i + 1];
                    }
                    n--;
                    cout << "Element deleted successfully.\n";
                }
                break;

            case 6:
                for (i = 0; i < n - 1; i++) {
                    for (int j = 0; j < n - i - 1; j++) {
                        if (arr[j] > arr[j + 1]) {
                            int temp = arr[j];
                            arr[j] = arr[j + 1];
                            arr[j + 1] = temp;
                        }
                    }
                }
                cout << "Sorted array: ";
                for (i = 0; i < n; i++) {
                    cout << arr[i] << " ";
                }
                cout << endl;
                break;

            default:
                cout << "Invalid choice.\n";
        }
    }
    return 0;
}
